test/built-ins/Temporal/Calendar/prototype/monthDayFromFields/fields-missing-properties.js

```diff
@@ -10,15 +10,19 @@ info: |
   3. Assert: calendar.[[Identifier]] is "iso8601".
   4. If Type(fields) is not Object, throw a TypeError exception.
   5. Set options to ? GetOptionsObject(options).
-  6. Let result be ? ISOMonthDayFromFields(fields, options).
-  7. Return ? CreateTemporalMonthDay(result.[[Month]], result.[[Day]], calendar, result.[[ReferenceISOYear]]).
+  6. Set fields to ? PrepareTemporalFields(fields, « "day", "month", "monthCode", "year" », « "day" »).
+  7. Let overflow be ? ToTemporalOverflow(options).
+  8. Perform ? ISOResolveMonth(fields).
+  9. Let result be ? ISOMonthDayFromFields(fields, overflow).
+  10. Return ? CreateTemporalMonthDay(result.[[Month]], result.[[Day]], "iso8601", result.[[ReferenceISOYear]]).
 features: [Temporal]
 ---*/
 
 let cal = new Temporal.Calendar("iso8601")
 
 assert.throws(TypeError, () => cal.monthDayFromFields({}), "at least one correctly spelled property is required");
+assert.throws(TypeError, () => cal.monthDayFromFields({ month: 12 }), "day is required with month");
 assert.throws(TypeError, () => cal.monthDayFromFields({ monthCode: "M12" }), "day is required with monthCode");
 assert.throws(TypeError, () => cal.monthDayFromFields({ year: 2021, month: 12 }), "day is required with year and month");
-assert.throws(TypeError, () => cal.monthDayFromFields({ month: 1, day: 17 }), "year is required if month is present");
```

Comment:
This assertion is no longer appropriate because `monthDayFromFields({ month, day })` is acceptable in the ISO 8601 calendar.

Reply:
(case 2.ii)